src/main/webapp/app/course/competencies/forms/common-course-competency-form.component.html

```diff
@@ -0,0 +1,135 @@
+                <div class="alert alert-danger">
+                    @if (descriptionControl?.errors?.maxlength) {
+                        <div
+                            [jhiTranslate]="'artemisApp.competency.' + competencyType + '.descriptionMaxLengthValidationError'"
+                            [translateValues]="{ max: competencyValidators.DESCRIPTION_MAX }"
+                        ></div>
+                    }
+                </div>
+            }
+        </div>
+    }
+    <div class="form-group">
+        <jhi-date-time-picker
+            id="softDueDate"
+            labelName="{{ 'artemisApp.' + competencyType + '.create.softDueDate' | artemisTranslate }}"
+            labelTooltip="{{ 'artemisApp.' + competencyType + '.create.softDueDateHint' | artemisTranslate }}"
+            formControlName="softDueDate"
+        />
+    </div>
+    @if (!isInConnectMode) {
+        <div class="form-group">
+            <label for="taxonomy">
+                {{ 'artemisApp.courseCompetency.taxonomy' | artemisTranslate }}
+                @if (suggestedTaxonomies?.length) {
+                    <small jhiTranslate="artemisApp.courseCompetency.create.suggestedTaxonomy"></small>
+                }
+            </label>
+            <jhi-taxonomy-select selectId="taxonomy" [form]="taxonomyControl" />
+        </div>
+    }
+    @if (!isInConnectMode) {
+        <div class="form-group">
+            <label for="masteryThreshold">
+                {{ 'artemisApp.courseCompetency.masteryThreshold' | artemisTranslate }}: {{ masteryThresholdControl!.value ?? DEFAULT_MASTERY_THRESHOLD }}%
+                @if (averageStudentScore) {
+                    <small> ({{ 'artemisApp.courseCompetency.create.averageMastery' | artemisTranslate }}: {{ averageStudentScore }}%) </small>
+                }
+            </label>
+            <input type="range" min="0" max="100" class="form-range" id="masteryThreshold" formControlName="masteryThreshold" />
+        </div>
+    }
+    <div class="form-group">
+        <input type="checkbox" class="form-check-input" id="optional" formControlName="optional" />
+        <label class="form-control-label" [jhiTranslate]="'artemisApp.' + competencyType + '.optional'" for="optional"></label>
+        <jhi-help-icon [text]="'artemisApp.' + competencyType + '.optionalDescription'" />
+    </div>
+    <div>
+        <label [jhiTranslate]="'artemisApp.' + competencyType + '.create.connectWithLectureUnits'"></label>
+        @if (lecturesOfCourseWithLectureUnits && lecturesOfCourseWithLectureUnits.length > 0) {
+            <div [hidden]="isInSingleLectureMode" ngbDropdown class="mb-2">
+                <button class="btn btn-outline-primary" ngbDropdownToggle type="button">
+                    {{
+                        selectedLectureInDropdown ? getLectureTitleForDropdown(selectedLectureInDropdown) : ('artemisApp.courseCompetency.create.selectLecture' | artemisTranslate)
```

Comment:
replace artemisTranslate pipe

Reply:
How would you do it? We only use the pipe for the second argument. The return of `getLectureTitleForDropdown` is not translated

Reply:
I would use a span like
```
<span [jhiTranslate]="selectedLectureInDropdown ? getLectureTitleForDropdown() : 'artemisApp.courseCompetency.create.selectLecture')
```
You may also have to remove the pipe then also from th getLectureTitleForDropown() method

Reply:
This will not work. You would get a translation not found in case `getLectureTitleForDropdown()` is used, since it already returns a translated string